src/validationinterface.cpp

```diff
@@ -80,17 +99,125 @@ void ThreadNotifyWallets()
 
         boost::this_thread::interruption_point();
 
+        auto chainParams = Params();
+
+        //
         // Collect all the state we require
+        //
+
+        // The common ancestor between the last chain tip we notified and the
+        // current chain tip.
+        const CBlockIndex *pindexFork;
+        // The stack of blocks we will notify as having been connected.
+        // Pushed in reverse, popped in order.
+        std::vector<CachedBlockData> blockStack;
+        // Transactions that have been recently conflicted out of the mempool.
+        std::map<CBlockIndex*, std::list<CTransaction>> recentlyConflicted;
+        // Transactions that have been recently added to the mempool.
+        std::pair<std::vector<CTransaction>, uint64_t> recentlyAdded;
+
+        {
+            LOCK(cs_main);
+
+            // Figure out the path from the last block we notified to the
+            // current chain tip.
+            CBlockIndex *pindex = chainActive.Tip();
+            pindexFork = chainActive.FindFork(pindexLastTip);
+
+            // Fetch recently-conflicted transactions. These will include any
+            // block that has been connected since the last cycle, but we only
+            // notify for the conflicts created by the current active chain.
```

Comment:
@daira would like a note about this in the ZMQ documentation (which is the only supported place where we might conceivably observe a change in behaviour).